src/runtime/client-event-manager-test.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+import * as EventManagerApi from '../api/client-event-manager-api';
+import {ClientEventManager} from './client-event-manager';
+
+const DEFAULT_TYPE = AnalyticsEvent.IMPRESSION_AD;
+const DEFAULT_ORIGIN = EventOriginator.SWG_CLIENT;
+const OTHER_TYPE = AnalyticsEvent.ACTION_PAYMENT_COMPLETE;
+const OTHER_ORIGIN = EventOriginator.AMP_CLIENT;
+const BAD_VALUE = 'I should throw an error';
+
+/** @type {!EventManagerApi.ClientEvent} */
+const DEFAULT_EVENT = {
+  eventType: DEFAULT_TYPE,
+  eventOriginator: DEFAULT_ORIGIN,
+  isFromUserAction: null,
+  additionalParameters: {},
+};
+
+/**
+ * Returns a standard string used to describe an issue with an event object
+ * @param {!string} valueName
+ * @param {?*} value
+ * @returns {!string}
+ */
+function createEventErrorMessage(valueName, value) {
```

Comment:
The catch here is that the same code is being used in test and in prod. So we won't ever know what to expect.  Wherever there's expected message, why not just define a mannual string. You are making a call to mannually create the expected message anyway.

Reply:
For the same reason you functionalize things for the production code: so if you change it you only need to change it once. It's now twice (once for the actual code and once for the test code) - but at least it isn't 8 times that then grows linearly every time you add another property.

Reply:
Though truthfully I don't understand your comment "we won't ever know what to expect".

Reply:
when test code duplicates production code, it's not clear what to expect. i'd recommend removing expected message helper in the test and actually spell out the entire string. This makes it clear what is expected and how the implementation accomplishes that is not necessary for the test to know.

Reply:
It just means that if the error message is changed it needs to be updated in multiple locations but OK